samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,293 @@
+
+        JsonPath jsonPath = RestAssured
+            .given()
+                .headers(API_KEY_HEADER_KEY, API_KEY_HEADER_VALUE)
+                .contentType(ContentType.JSON)
+                .body(sampleDataRequest)
+            .when()
+                .post(INITIATE_TRANSFER_PROCESS_URI)
+            .then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract()
+                .jsonPath();
+
+        String transferProcessId = jsonPath.get("id");
+
+        assertThat(transferProcessId).isNotEmpty();
+    }
+
+    /**
+     * Reads a transfer request file with changed value for contract agreement ID and file destination path.
+     * @param transferJsonFile A {@link File} instance pointing to a JSON transfer request file.
+     * @param contractAgreementId This string containing a UUID will be used as value for the contract agreement ID.
+     * @return An instance of {@link DataRequest} with changed values for contract agreement ID and file destination path.
+     * @throws IOException Thrown if there was an error accessing the file given in transferJsonFile.
+     */
+    DataRequest readAndUpdateTransferJsonFile(File transferJsonFile, String contractAgreementId) throws IOException {
+        // create object mapper instance
+        ObjectMapper mapper = new ObjectMapper();
+
+        // convert JSON file to map
+        DataRequest sampleDataRequest = mapper.readValue(transferJsonFile, DataRequest.class);
+
+        DataAddress newDataDestination = sampleDataRequest.getDataDestination().toBuilder().property("path", destinationFile.getAbsolutePath()).build();
+        DataRequest newSampleDataRequest = sampleDataRequest.toBuilder().contractId(contractAgreementId).dataDestination(newDataDestination).build();
+
+        return newSampleDataRequest;
+    }
+
+    /**
+     * Assert that the properties.config files are properly configured and all ports uniquely used.
+     * Relates to sample guidance section: Create the connectors.
+     */
+    void assertConfigPropertiesUniquePorts() {
+        String pathConsumerConfigProperties = getFileFromRelativePath(CONSUMER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath();
+        String pathProviderConfigProperties = getFileFromRelativePath(PROVIDER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath();
+
+        Properties consumerProperties = null;
+        Properties providerProperties = null;
+
+        try {
+            consumerProperties = readPropertiesFile(pathConsumerConfigProperties);
+            providerProperties = readPropertiesFile(pathProviderConfigProperties);
+        } catch (IOException e) {
+            fail("Could not read properties files.");
+        }
+
+        List<String> portPropertyNames = List.of(
+                "web.http.port",
+                "web.http.data.port",
+                "web.http.ids.port"
+        );
+
+        HashMap<Integer, String> configuredPorts = new HashMap<>();
```

Comment:
```suggestion
        Map<Integer, String> configuredPorts = new HashMap<>();
```

Or even better use `var`

Reply:
With `var` the types change to `HashMap<Object, Object>` which is not the desired result.

However, I have removed the test for unique ports as it runs too late in the life cycle of the test. The EDC runtime extensions are instantiated before the tests run. Therefor a reuse of ports will fail before the tests could fail.